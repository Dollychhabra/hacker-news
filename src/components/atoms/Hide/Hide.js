import React from 'react';
import PropTypes from 'prop-types';

import './Hide.scss';
import Anchor from '../Anchor';

const Hide = ({ hideData, objectId }) => {
  return (
    <Anchor
      className="hide-btn"
      handleClick={e => {
        e.preventDefault();
        hideData(objectId);
      }}
    >
      hide
    </Anchor>
  );
};

Hide.propTypes = {
  hideData: PropTypes.func.isRequired,
  objectId: PropTypes.string.isRequired,
};

export default Hide;
