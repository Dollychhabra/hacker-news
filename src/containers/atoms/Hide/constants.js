export const HIDE_DATA_KEY = 'HACKER_NEWS_APP/HOME_PAGE/HIDE_DATA_KEY';
export const HIDE_DATA = 'HACKER_NEWS_APP/HOME_PAGE/HIDE_DATA';
