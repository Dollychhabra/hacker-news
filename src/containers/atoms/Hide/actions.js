import { HIDE_DATA } from './constants';

export const hideData = data => ({
  type: HIDE_DATA,
  data,
});
