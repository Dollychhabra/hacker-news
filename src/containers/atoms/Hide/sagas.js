/* eslint-disable require-yield */
import { takeLatest } from 'redux-saga/effects';

import { HIDE_DATA, HIDE_DATA_KEY } from './constants';
import { LocalStorageUtil } from '../../../utils/localStorage';

export function* saveHideDataSaga(action) {
  const { data: objectId } = action;

  const storage = new LocalStorageUtil();

  storage.saveItemInArray(HIDE_DATA_KEY, objectId);
}

export default function* hideDataSaga() {
  yield takeLatest(HIDE_DATA, saveHideDataSaga);
}
