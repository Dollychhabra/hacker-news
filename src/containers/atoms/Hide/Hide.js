import { connect } from 'react-redux';

import { hideData } from './actions';

import Hide from '../../../components/atoms/Hide';

export const mapDispatchToProps = dispatch => {
  return {
    hideData: itemId => {
      dispatch(hideData(itemId));
    },
  };
};

export default connect(
  null,
  mapDispatchToProps,
)(Hide);
