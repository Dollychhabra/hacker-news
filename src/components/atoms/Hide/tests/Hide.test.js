import React from 'react';
import { shallow } from 'enzyme';
import Hide from '../Hide';

describe('Hide component', () => {
  const props = {
    hideData: () => {},
    objectId: '453534',
  };

  it('should render correctly and match snapshot', () => {
    const component = shallow(<Hide {...props} />);
    expect(component).toMatchSnapshot();
  });
});
