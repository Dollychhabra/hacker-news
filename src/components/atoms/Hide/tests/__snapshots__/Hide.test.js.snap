// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hide component should render correctly and match snapshot 1`] = `
<Anchor
  className="hide-btn"
  handleClick={[Function]}
  target="_self"
  to=""
>
  hide
</Anchor>
`;
